src/memory/AmdFlash.cc

```diff
@@ -442,7 +442,14 @@ uint16_t AmdFlash::peekCFI(size_t address) const
 
 bool AmdFlash::isWritable(const Sector& sector) const
 {
-	return vppWpPinLow && (sector == one_of(sectors[0], sectors[1])) ? false : !sector.writeProtect;
+	if (vppWpPinLow) {
+		const auto range = chip.misc.writeProtectPinRange;
+		if ((range > 0 && sector.address <  sectors[range + size_t(0)].address) ||
```

Comment:
Is it ever possible that `abs(writeProtectPinRange) == sectorCount`? It should never be strictly greater, but can it be equal?
If that's possible then there's an out-of-bounds access (for `range > 0`, `range < 0` happens to work).

Reply:
To my knowledge there is no hardware where the write protect pin protects the entire memory, but I’ll strengthen the checks and logic to make sure it doesn’t go out of bounds regardless.